### PR TITLE
added name of Var to error message

### LINKED ISSUE #2826: include the name of the Var so it's easier to track down `No JSON serializer` error
It will be nice to also include the name of the Var causing the ```TypeError(f"No JSON serializer found for var {value} of type {type(value)}.")```  so it's easier for a user to track down and maybe convert the Var to a string

### PR DESCRIPTION
**Pull request by emmakodes**
-  Added the name of the Var to the error message for ```TypeError(f"No JSON serializer found for var {value} of type {type(value)}.")``` closes #2826 

### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.
        
    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Comment by emmakodes** [comment]
@masenf

File: reflex/utils/format.py [hunk 1]
```diff
@@ -598,7 +600,14 @@ def format_state(value: Any) -> Any:
     """
     # Handle dicts.
     if isinstance(value, dict):
-        return {k: format_state(v) for k, v in value.items()}
+        result = {}
+        for k, v in value.items():
+            if isinstance(v, (pathlib.PosixPath, functools.partial)):
```
Comment (picklelo) [inline reflex/utils/format.py:605]: I think we can be more general here by either:
1. Using `reflex.utils.serailizers.has_serializer` to check
2. Just putting a try/except around the setting.

Actually I think the easiest fix would be to pass down the key to `format_state` and just update the error message thrown there.

**Review by picklelo — CHANGES_REQUESTED** [review]

File: reflex/utils/format.py (hunk 1 above)
Comment (emmakodes) [reply reflex/utils/format.py]: Thanks, @picklelo, your suggestion makes sense.

Comment (emmakodes) [reply reflex/utils/format.py]: I have now updated the code by passing the key to `format_state` and updating the error message

**Review by picklelo — APPROVED** [review]
awesome, much better messages now. thanks for adding!

**Pull request merged**